httptools/parser.pyi

```diff
@@ -0,0 +1,88 @@
+from typing import Optional
+
+
+class HttpRequestParser:
+    def __init__(self, protocol):
+        """HttpRequestParser
+
+        protocol -- a Python object with the following methods
+        (all optional):
+
+          - on_message_begin()
+          - on_url(url: bytes)
+          - on_header(name: bytes, value: bytes)
+          - on_headers_complete()
+          - on_body(body: bytes)
+          - on_message_complete()
+          - on_chunk_header()
+          - on_chunk_complete()
+          - on_status(status: bytes)
+        """
+        pass
```

Comment:
By [convention](https://mypy.readthedocs.io/en/stable/stubs.html#stub-file-syntax), stub function bodies should use `...` instead of `pass`.